client/extensions/woocommerce/app/settings/shipping/shipping-zone/shipping-zone-location-list.js

```diff
@@ -112,14 +112,16 @@ const ShippingZoneLocationList = ( { siteId, loaded, translate, locations, actio
 				</Button>
 			</ExtendedHeader>
 			<List>
-				<ListHeader>
+				{ locationsToRender && locationsToRender.length
```

Comment:
In the very next block you're using `locationsToRender.map()`. So that means that `locationsToRender` is guaranteed to be an array at this point.